src/commands/api/findTreeItem.ts

```diff
@@ -26,14 +27,29 @@ export async function findTreeItem(query: TreeItemQuery): Promise<DatabaseAccoun
         context.errorHandling.suppressDisplay = true;
         context.errorHandling.rethrow = true;
 
-        const connectionString = query.connectionString;
         let parsedCS: ParsedConnectionString;
-        if (/^mongodb[^:]*:\/\//i.test(connectionString)) {
-            parsedCS = await parseMongoConnectionString(connectionString);
-        } else if (/^postgres:\/\//i.test(connectionString)) {
+        if (query.postgresData) {
+            const postgresData = query.postgresData;
+            let connectionString: string = `postgres://`;
```

Comment:
This should leverage `createPostgresConnectionString`. Also, `createPostgresConnectionString` and `getEncodedConnectionString` should be combined somehow. It's all the same logic for creating a connection string